### PR TITLE
chore: cleanup Pr for multiple-environment

### PR DESCRIPTION
**Pull request by sondermanish**
Fixes #https://github.com/appsmithorg/appsmith-ee/issues/1507, https://github.com/appsmithorg/appsmith/issues/25604

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?

- [ ] Manual

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration020TransferToDatasourceStorage.java [hunk 1]
```diff
@@ -154,4 +154,18 @@ private Criteria findDatasourceIdsToUpdate() {
                                         + PASSWORD)
                                 .ne(""));
     }
+
+    private static DatasourceStorage createDatasourceStorageFromDatasource(
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration020TransferToDatasourceStorage.java:158]: Why static? 🤔

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration020TransferToDatasourceStorage.java [hunk 2]
```diff
@@ -154,4 +154,18 @@ private Criteria findDatasourceIdsToUpdate() {
...
+            Datasource datasource, String environmentId) {
+        DatasourceStorage datasourceStorage = new DatasourceStorage(
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration020TransferToDatasourceStorage.java:160]: Should we be adding validations at this point? Can be taken up as a separate task if you agree.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration020TransferToDatasourceStorage.java (hunk 1 above)
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration020TransferToDatasourceStorage.java:158]: No specific reason for this, earlier this method declared in DatasourceStorageService, just a copy over as the other one was made non static

**Review by nidhi-nair — APPROVED** [review]

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**